src/mono/msbuild/apple/build/AppleBuild.LocalBuild.targets

```diff
@@ -25,17 +25,82 @@
   <UsingTask TaskName="RuntimeConfigParserTask" AssemblyFile="$(MonoTargetsTasksAssemblyPath)" />
 
   <PropertyGroup>
+    <GenerateXcodeProject>true</GenerateXcodeProject>
+    <GenerateCMakeProject>false</GenerateCMakeProject>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="'$(EnableAggressiveTrimming)' != 'true' and '$(EnableSoftTrimming)' != 'true'">
     <PublishTrimmed>true</PublishTrimmed>
     <TrimMode>link</TrimMode>
   </PropertyGroup>
+  <PropertyGroup Condition="'$(EnableAggressiveTrimming)' == 'true' or '$(EnableSoftTrimming)' == 'true'">
+    <PublishTrimmed>true</PublishTrimmed>
+    <!-- Suppress trimming warnings as these are tests -->
+    <SuppressTrimAnalysisWarnings>true</SuppressTrimAnalysisWarnings>
+    <NoWarn>$(NoWarn);IL2103;IL2105;IL2025;IL2111</NoWarn>
+    <!-- Warns about missing Xamarin.iOS.dll -->
+    <NoWarn Condition="'$(EnableSoftTrimming)' == 'true'">$(NoWarn);IL2035</NoWarn>
+    <!-- https://github.com/dotnet/sdk/issues/18581 tracks needing to set 2 properties to disable warnings. -->
+    <EnableTrimAnalyzer>false</EnableTrimAnalyzer>
 
-  <PropertyGroup>
-    <GenerateXcodeProject>true</GenerateXcodeProject>
-    <GenerateCMakeProject>false</GenerateCMakeProject>
+    <!-- Reduce library test app size by trimming framework library features  -->
+    <DebuggerSupport Condition="'$(DebuggerSupport)' == '' and '$(Configuration)' != 'Debug'">false</DebuggerSupport>
+    <EnableUnsafeUTF7Encoding Condition="'$(EnableUnsafeUTF7Encoding)' == ''">false</EnableUnsafeUTF7Encoding>
+    <EventSourceSupport Condition="'$(EventSourceSupport)' == ''">false</EventSourceSupport>
+    <HttpActivityPropagationSupport Condition="'$(HttpActivityPropagationSupport)' == ''">false</HttpActivityPropagationSupport>
+    <UseSystemResourceKeys Condition="'$(UseSystemResourceKeys)' == ''">false</UseSystemResourceKeys>
+    <UseNativeHttpHandler Condition="'$(UseNativeHttpHandler)' == ''">false</UseNativeHttpHandler>
```

Comment:
@kotlarmilos me might want to reconsider this, as my last change recreates the state of `RuntimeConfigurationOption` from the build-time to helix-build-time.
The same applies to this change: https://github.com/dotnet/runtime/pull/96169/files#diff-b9f66b9752f1aa02d62fa3062d8e57ba707c40653941ac33ad7c87dbcf9f5597R220

Reply:
I agree, then we don't need these props. Also, I suggest we revert the `IsDebuggerTypeProxyAttributeSupported` change.